File: app/src/main/java/com/ptrprograms/machinelearningdemo/TextRecognition/TextRecognitionActivity.java
package com.ptrprograms.machinelearningdemo.TextRecognition;

import android.hardware.camera2.CameraAccessException;
import android.media.Image;
import android.media.ImageReader;
import android.support.annotation.NonNull;
import android.util.Log;
import android.widget.ImageView;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.ml.vision.FirebaseVision;
import com.google.firebase.ml.vision.common.FirebaseVisionImage;
import com.google.firebase.ml.vision.text.FirebaseVisionText;
import com.google.firebase.ml.vision.text.FirebaseVisionTextDetector;
import com.ptrprograms.machinelearningdemo.R;
import com.ptrprograms.machinelearningdemo.camera.CameraActivity;

public class TextRecognitionActivity extends CameraActivity {

    protected ImageView imageView;

    @Override
    protected void initViews() {
        imageView = findViewById(R.id.image);
    }

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_camera;
    }

    @Override
    public void onImageAvailable(ImageReader reader) {
        Log.e("Test", "image available");
        final Image image = reader.acquireNextImage();
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                imageView.setImageBitmap(convertImageToBitmap(image));
            }
        });
        int rotation = 0;

        try {
            rotation = getRotationCompensation(this);
        } catch( CameraAccessException e ) {
            Log.e("Machine Learning", "camera access exception");
        }

        FirebaseVisionImage firebaseVisionImage = FirebaseVisionImage.fromMediaImage(image, rotation);

        FirebaseVisionTextDetector detector = FirebaseVision.getInstance()
                .getVisionTextDetector();

        Task<FirebaseVisionText> result =
                detector.detectInImage(firebaseVisionImage)
                        .addOnSuccessListener(new OnSuccessListener<FirebaseVisionText>() {
                            @Override
                            public void onSuccess(FirebaseVisionText firebaseVisionText) {
                                Log.e("Machine Learning", "success!");

                                for( FirebaseVisionText.Block block : firebaseVisionText.getBlocks() ) {
                                    Log.e("Machine Learning", block.getText() );
                                }
                            }
                        })
                        .addOnFailureListener(
                                new OnFailureListener() {
                                    @Override
                                    public void onFailure(@NonNull Exception e) {
                                        Log.e("Machine Learning", "failure :( " + e.getMessage());
                                    }
                                });

        setReady(true);

        reader.close();

        resetCamera();
    }

}

File: app/src/main/java/com/ptrprograms/machinelearningdemo/TextRecognition/TextRecognitionActivity.kt
package com.ptrprograms.machinelearningdemo.TextRecognition

import android.hardware.camera2.CameraAccessException
import android.media.ImageReader
import android.util.Log
import com.google.firebase.ml.vision.FirebaseVision
import com.google.firebase.ml.vision.common.FirebaseVisionImage
import com.ptrprograms.machinelearningdemo.R
import com.ptrprograms.machinelearningdemo.camera.CameraActivity
import kotlinx.android.synthetic.main.activity_camera.*

class TextRecognitionActivity(override val layoutResId: Int = R.layout.activity_camera) : CameraActivity() {

    override fun onImageAvailable(reader: ImageReader) {
        Log.e("Test", "image available")
        val image = reader.acquireNextImage()
        runOnUiThread { camera_image.setImageBitmap(convertImageToBitmap(image)) }
        var rotation = 0

        try {
            rotation = getRotationCompensation(this)
        } catch (e: CameraAccessException) {
            Log.e("Machine Learning", "camera access exception")
        }

        val firebaseVisionImage = FirebaseVisionImage.fromMediaImage(image, rotation)

        val detector = FirebaseVision.getInstance()
                .visionTextDetector

        val result = detector.detectInImage(firebaseVisionImage)
                .addOnSuccessListener { firebaseVisionText ->
                    Log.e("Machine Learning", "success!")

                    var blocks = ArrayList<String>()
                    for (block in firebaseVisionText.blocks) {
                        Log.e("Machine Learning", block.text)
                        blocks.add(block.text)
                    }

                    camera_text.setText(blocks.joinToString())
                }
                .addOnFailureListener { e -> Log.e("Machine Learning", "failure :( " + e.message) }

        setReady(true)

        reader.close()

        resetCamera()
    }

}
